ledger/trackerdb.go

```diff
@@ -101,10 +110,10 @@ func trackerDBInitialize(l ledgerForTracker, catchpointEnabled bool, dbPathPrefi
 	return
 }
 
-// trackerDBInitializeImpl initializes the accounts DB if needed and return current account round.
+// runMigrations initializes the accounts DB if needed and return current account round.
 // as part of the initialization, it tests the current database schema version, and perform upgrade
 // procedures to bring it up to the database schema supported by the binary.
-func trackerDBInitializeImpl(ctx context.Context, tx *sql.Tx, params trackerDBParams, log logging.Logger) (mgr trackerDBInitParams, err error) {
+func runMigrations(ctx context.Context, tx *sql.Tx, params trackerDBParams, log logging.Logger, targetVersion int32) (mgr trackerDBInitParams, err error) {
```

Comment:
Probably not useful for this PR, but maybe in the future it would be.

There are some utilities added with the participation registry to fill in the boilerplate for these sorts of migrations. You pass an array of `Migration` functions and it handles the versioning and calling the correct subset of migration handlers.

https://github.com/algorand/go-algorand/blob/master/util/db/initialize.go#L34

At one point I converted the accountDB to use the new utilities, it was mostly straightforward:
https://github.com/winder/go-algorand/pull/3/files#

Reply:
We have a similar problem with the REST handlers that was solved by making handlers receiver functions and putting the extra state in a config struct.

Closure functions sometimes work nicely for this also:
```
	// Wrapper functions to coerce the existing migration functions into the new interface.
	// TODO: update upgradeDatabaseSchema* functions instead of having wrappers. This was
	//       skipped to simplify the code review.
	wrapper := func(schemaFunc func(ctx context.Context, tx *sql.Tx, newDatabase bool) (int32, error)) db.Migration {
		return func(ctx context.Context, tx *sql.Tx, newDatabase bool) error {
			_, err := schemaFunc(ctx, tx, newDatabase)
			return err
...
		wrapper(au.upgradeDatabaseSchema1),
		wrapper(au.upgradeDatabaseSchema2),
		wrapper(au.upgradeDatabaseSchema3),
		wrapper(au.upgradeDatabaseSchema4),
```

Maybe something like that would work here?